### PR TITLE
Lfs fix

### PR DESCRIPTION
**Pull request by langausd**
made LFS work again with current nodemcu versions.
(tried to keep fallbacks to old versions, but unable to test)

### CODE REVIEW

**Comment by marcoskirsch** [comment]
>(tried to keep fallbacks to old versions, but unable to test)

I'm perfectly fine with `htnodemcu-httpserver` working only with latest firmware. It's too complex to support old firmware versions.

File: httpserver-conf.lua [hunk 1]
```diff
@@ -13,7 +13,7 @@ conf.general.port = 80
 -- WiFi configuration
 conf.wifi = {}
 -- Can be wifi.STATION, wifi.SOFTAP, or wifi.STATIONAP
-conf.wifi.mode = wifi.STATION
+conf.wifi.mode = wifi.SOFTAP
```
Comment (marcoskirsch) [inline httpserver-conf.lua]: this change seems unrelated, can you revert? (or if related please explain)

File: init.lua [hunk 2]
```diff
@@ -9,12 +9,19 @@ if node.getpartitiontable().lfs_size > 0 then
...
 	 f:close()
 	 file.remove("httpserver-compile.lua")
-	 node.flashreload("lfs.img")
+	 if (node.LFS.reload) then
+	    node.LFS.reload("lfs.img")
+	 else
+	    print "using old flashreload fallback."
+	    print " Consider updating to recent nodeMCU firmware!" 
+	    node.flashreload("lfs.img")
+	 end
```
Comment (marcoskirsch) [inline init.lua]: I don't think this is worth keeping around. Especially if it's going to be mostly an untested codepath.

**Review by marcoskirsch — CHANGES_REQUESTED** [review]
Thanks for updating the server code!

File: init.lua (hunk 2 above)
Comment (langausd) [reply init.lua]: I guess you're right.
However, the readme should reflect that current (3.0.0 or newer?) nodemcu-firmware is required.
BTW, I'm also testing a stripped down LFS only version here: https://github.com/langausd/nodemcu-httpserver/tree/LFS-framework

That does definitely break compatibility, but really performs nicely (while serving almost 2 counter.html instances per second, it's even able to serve all six cars wihout crashing...)

File: httpserver-conf.lua (hunk 1 above)
Comment (langausd) [reply httpserver-conf.lua]: yup, it's unrelated. Just is a working default, as opposed to the station config that will result in a reboot loop, unless you have an open wifi named "Internet" ;-)
If you prefer a separate pull request, that's fine with me.

**Review by langausd — COMMENTED** [review]
changes should be applied now.
Sorry for the delay. I'm quite new to github, and itseems my github notifications are eaten by spamfilter...

**Review by marcoskirsch — APPROVED** [review]
